libraries/engage/product/components/FulfillmentSelector/index.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import { SurroundPortals } from '@shopgate/engage/components';
+import I18n from '@shopgate/pwa-common/components/I18n';
+import RadioGroup from '@shopgate/pwa-ui-shared/Form/RadioGroup';
+import RadioItem from '@shopgate/pwa-ui-shared/Form/RadioGroup/components/Item';
+import { isBeta } from '../../../core/config/isBeta';
+import {
+  PRODUCT_FULFILLMENT_METHOD_DIRECT_SHIP,
+  PRODUCT_FULFILLMENT_METHOD_IN_STORE_PICKUP,
+} from '../../constants';
+import { PRODUCT_FULFILLMENT_SELECTOR } from '../../constants/Portals';
+import Availability from '../Availability';
+import { StoreSelector } from '../StoreSelector';
+import LocationStockInfo from '../LocationStockInfo';
+import connect from './connector';
+import * as styles from './style';
+
+// TODO: Integrate a conditioner that ensures selection of characteristics and required options
+/**
+ * Fulfillment/Store selector box
+ * @return {JSX}
+ */
+export const FulfillmentSelector = ({ productCode, fulfillmentMethods, location }) => {
+  if (!isBeta()) { // TODO: Change to feature flag!
+    return null;
+  }
+
+  // Don't render, when no pick up in store is available for the given product.
+  if (!fulfillmentMethods
+    || fulfillmentMethods.indexOf(PRODUCT_FULFILLMENT_METHOD_IN_STORE_PICKUP) === -1) {
+    return null;
+  }
+
+  const directShip = 'product.fulfillment_selector.direct_ship';
+  const pickUp = 'product.fulfillment_selector.pick_up_in_store';
+
+  // Pre-select direct ship on entering the PDP
+  const [selection, setSelection] = useState(directShip);
+
+  // Keep the selected location in the state
+  const [selectedLocation, setSelectedLocation] = useState(null);
+
+  // Keeps track of the selector sheet being opened or not, to "debounce" open events.
+  const [isSelectorOpened, setIsSelectorOpened] = useState(false);
+
+  // When the product code changes, the selection will reset
+  const [currentProductCode, setCurrentProductCode] = useState(productCode);
+  if (productCode !== currentProductCode) {
```

Comment:
I wouldn't call it dangerous, since this is pretty much how `useEffect` works internally (a state combined with a comparison to props).
But you're right that is easier to understand when the actual `useEffect` functionality is used over a custom solution. I'll change it.